pos_sale_order/point_of_sale.py

```diff
@@ -0,0 +1,239 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    POS To Sale Order module for Odoo
+#    Copyright (C) 2014 AKRETION (<http://www.akretion.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import time
+
+from openerp import fields, models, api, _
+from openerp.exceptions import Warning
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    pos_reference = fields.Char(string='Receipt Ref',
+                                readonly=True,
+                                copy=False,
+                                default='')
+    session_id = fields.Many2one('pos.session', string='Session',
+                                 select=1,
+                                 domain="[('state', '=', 'opened')]",
+                                 states={'draft': [('readonly', False)]},
+                                 readonly=True)
+    payment_ids = fields.Many2many(readonly=True)
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    @api.multi
+    def _prepare_sale_order_vals(self, ui_order):
+        sale_line_obj = self.env['sale.order.line'].browse(False)
+        if not ui_order['partner_id']:
+            session = self.env['pos.session'].browse(
+                ui_order['pos_session_id'])
+            partner_id = session.config_id.anonymous_partner_id
+            ui_order['partner_id'] = partner_id
+        for line in ui_order['lines']:
+            if line[2].get('qty'):
+                line[2]['product_uom_qty'] = line[2].pop('qty')
+            defaults = sale_line_obj.product_id_change(
+                pricelist=False,
```

Comment:
As we know the pricelist (in the config) it we be better to pass the right pricelist